js/src/common/utils/extractText.ts

```diff
@@ -1,15 +1,23 @@
 /**
  * Extract the text nodes from a virtual element.
  *
- * @param {VirtualElement} vdom
- * @return {String}
+ * @param {IVirtualElement} vdom
+ * @return {string}
  */
-export default function extractText(vdom) {
+export default function extractText(vdom: IVirtualElement): string {
   if (vdom instanceof Array) {
     return vdom.map((element) => extractText(element)).join('');
   } else if (typeof vdom === 'object' && vdom !== null) {
+    // @ts-ignore
     return vdom.children ? extractText(vdom.children) : vdom.text;
```

Comment:
Would be preferable to avoid ts-ignore, I believe using mithril typing would give us consistency and fix the issue

Reply:
What error exactly is being ignored here with `ts-ignore`?

Reply:
On using `mithril.VnodeDOM` receiving `Type 'string' is not assignable to type 'VnodeDOM<{}, {}>'.ts(2345)` for the expression, `extractText(vdom.children)`.